amethyst_renderer/src/sprite.rs

```diff
@@ -24,19 +27,38 @@ pub struct SpriteSheet {
 }
 
 impl Asset for SpriteSheet {
-    const NAME: &'static str = "renderer::Sprite";
+    const NAME: &'static str = "renderer::SpriteSheet";
     type Data = Self;
     type HandleStorage = VecStorage<Handle<Self>>;
 }
 
-/// A description of a frame in a sprite sheet.
+impl From<SpriteSheet> for AssetsResult<ProcessingState<SpriteSheet>> {
+    fn from(sprite_sheet: SpriteSheet) -> AssetsResult<ProcessingState<SpriteSheet>> {
+        Ok(ProcessingState::Loaded(sprite_sheet))
+    }
+}
+
+/// Dimensions and texture coordinates of each sprite in a sprite sheet.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Sprite {
+    /// Pixel width of the sprite
+    pub width: f32,
+    /// Pixel height of the sprite
+    pub height: f32,
+    /// Number of pixels to shift the sprite to the left and down relative to the entity
+    pub offsets: [f32; 2],
```

Comment:
What will this be used for? If it is used to change the entity's origin, it might be better to have a parent entity with the sprite as a child entity + offset.

Reply:
"change the entity's origin" is sounds like it (it's, "change the origin of the sprite relative to the entity"). I wouldn't advocate for the parent with child entity + offset because:

* ergonomics: for games like a space shooter where one entity has one sprite, the consumer shouldn't have to create 2 entities just to offset the sprite rendering
* mirroring sprites: if using a parent + child entity, and they have a left/right mirror, they inadvertently have to code some graphics logic to calculate the flipped coordinates for the child entity, when what they really want to do is say `is_flipped: bool`